style_guide/README.md

```diff
@@ -0,0 +1,206 @@
+
+        git add <new or changed files you want to commit>
+        git commit -m 'Explaination of the changes'
+
+8. Push the feature branch to your personal fork's GitHub repository:
+
+        git push origin <branchname>
+
+9. On the `IMAP-Science-Operations-Center` `imap_processing`
+   [repository](https://github.com/IMAP-Science-Operations-Center/imap_processing)
+   create a new pull request.  Click on the "compare across forks" link to
+   enable the pull request to use your fork.  Set the "base repository" to
+   `IMAP-Science-Operations-Center` and "base" to `dev`.  Set the "head
+   repository" to the `imap_processing` repository under your personal fork
+   and "compare" to your feature branch. If the feature branch is still under
+   development, you can click the "Convert to draft" button under the
+   "Reviewers" section, or add a "[WIP]" at the beginning of the pull request
+   title to signify that the pull request is not ready to be merged.
+
+10. Assign at least one reviewer to the pull request.  They will review your
+    pull request and either accept the request or ask for additional changes.
+    If additional changes are needed, iterate through steps (6) and (7) until
+    you and the reviewer(s) are satisfied.
+
+11. Once the pull request has been accepted, you can merge the pull request and
+    delete the feature branch.
+
+
+### Making hotfixes
+
+As shown in the diagram above, sometimes hotfixes need to be made directly to
+the `main` branch. The workflow for this scenario is as follows:
+
+1. Assuming steps (1) through (3) in the previous section are already completed,
+   create a new branch named `hotfix-<description>` off of the `main` branch,
+   and commit any necessary changes following the nominal `git add`/`git commit`
+   cycle.
+2. Push the hotfix branch to your personal fork, and open two separate pull
+   requests: one that merges the hotfix branch into the
+   `IMAP-Science-Operations-Center` `imap_processing` `main` branch, and one
+   that merges the hotfix branch into the `IMAP-Science-Operations-Center`
+   `imap_processing` `dev` branch.
+3. For each of these pull requests, proceed with the nominal review & merge
+   process described in steps (9) and (10) in the previous section.
+
+
+### Keeping your fork updated
+
+You can keep your personal fork up-to-date with the
+`IMAP-Science-Operations-Cetner` `imap_processing` repository by fetching and
+pulling the `upstream` remote:
+
+    git checkout dev
+    git fetch upstream dev
+    git pull upstream/dev
+
+### Collaborating on someone else's fork
+
+To contribute to a branch on another person's personal fork, add a new `remote`
+that points to their fork, and use the nominal workflow for contributing:
+
+    git remote add <username> <remote url>
+    git fetch <username>
+    git checkout -b <branchname> <username>/<branchname>
+    # Make some changes via add/commit cycle
+    git push <username> <branchname>
+
+## Poetry Environment
+
+TBD
```

Comment:
```suggestion
Poetry is used for dependency management within this project. To update dependencies, you can either update `pyproject.toml` manually, or use `poetry add <dependency>`.

  If you do add dependencies, please make sure you define the version numbers mindfully. The best way to do this is to use [caret notation](https://python-poetry.org/docs/dependency-specification/#caret-requirements) to allow for minor version updates. 

For example, if you have tested the code using numpy version 1.24.2, the best way to specify the dependency is to allow for updates to the left most version number:

`numpy = "^1.24"`

This will allow for patches for security reasons without upgrading to 1.25 or beyond. Use your best judgement for when packages can upgrade automatically, but try and avoid specifying only one specific version unless it is absolutely required. 

The `poetry.lock` file contains the existing dependencies for the project. These are the dependencies that you should install to ensure you're getting the accepted versions for every package. If the poetry.lock file is not up to date, there is an automatic process within the repo to update poetry.lock to match `pyproject.toml`. To resolve any dependency issues before the precommit hook, users should run `poetry lock` and resolve any dependency problems ahead of time. The `poetry.lock` file should be part of the files committed in this case. 


```

Reply:
Awesome, thank you!